test_crates/enum_no_repr_variant_discriminant_changed/new/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// Explicit discriminant changed values. By doing so, it changed the implicit
+// discriminant's value as well, should be reported.
+pub enum ExplicitAndImplicitDiscriminantsAreChanged {
+    First = 2,
+    Second,
+    Third = 5,
+}
+
+// Discriminant changed values. Having #[non_exhaustive] on the enum should not have any effect
+// on the *API* breakage, should still be reported.
+#[non_exhaustive]
+pub enum DiscriminantIsChanged {
+    First = 1,
+}
+
+// Discriminant changed values and the variant is also marked as `non_exhaustive`.
+// This now implies that the enum is no longer `well-defined`, which means that a numeric
+// cast is no longer possible on the enum, should not be reported.
+// https://github.com/rust-lang/reference/pull/1249#issuecomment-2339003824
+#[non_exhaustive]
+pub enum DiscriminantIsChangedAndMarkedNonExhaustive {
+    First,
+    #[non_exhaustive]
+    Second = 2,
+}
+
+// Discriminant changed values, but the variant is already `non_exhaustive`.
+// This means that the enum is already not `well-defined`, and the numeric cast
+// was never possible, should not be reported.
+#[non_exhaustive]
+pub enum DiscriminantIsChangedButAlreadyNonExhaustive {
+    First,
+    #[non_exhaustive]
+    Second = 2,
+}
+
+// Discriminant changed to be doc hidden and explicit. Being doc hidden is not relevant
```

Comment:
Some more imprecise language:
```suggestion
// Variant became doc(hidden) while variant became explicit. Being doc(hidden) is not relevant
```